From 6c81df5916cacdf09345618ad3616d10c6c0e00f Mon Sep 17 00:00:00 2001
From: Alex Deucher <alexander.deucher@amd.com>
Date: Thu, 17 Aug 2017 16:41:13 -0400
Subject: [PATCH 2827/2831] drm/amdgpu: fix a bogus warning

Don't validate the default value.  Prevents a needless
warning. Also fix a spelling typo in the warning message.

Reviewed-by: Alex Xie <AlexBin.Xie@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_device.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_device.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_device.c
index 47db2e0..54b50d9 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_device.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_device.c
@@ -1101,8 +1101,9 @@ static void amdgpu_check_arguments(struct amdgpu_device *adev)
 	}
 
 	/* valid range is between 4 and 9 inclusive */
-	if (amdgpu_vm_fragment_size > 9 || amdgpu_vm_fragment_size < 4) {
-		dev_warn(adev->dev, "valid rang is between 4 and 9\n");
+	if (amdgpu_vm_fragment_size != -1 &&
+	    (amdgpu_vm_fragment_size > 9 || amdgpu_vm_fragment_size < 4)) {
+		dev_warn(adev->dev, "valid range is between 4 and 9\n");
 		amdgpu_vm_fragment_size = -1;
 	}
 
-- 
2.7.4

